src/module/actor/base.ts

```diff
@@ -722,6 +728,22 @@ class ActorPF2e<TParent extends TokenDocumentPF2e | null = TokenDocumentPF2e | n
         }
     }
 
+    protected prepareActions(): void {
+        this.actions = new ActorActions();
+        for (const action of this.synthetics.actions) {
+            const proxy = new Proxy(action, new ActionProxyHandler(this));
+            if (action.traits.includes("downtime")) {
+                this.actions.downtime.set(proxy.slug, proxy);
+            }
+            if (action.traits.includes("exploration")) {
+                this.actions.exploration.set(proxy.slug, proxy);
+            }
+            if (!!action.cost || action.variants.find((variant) => !!variant.cost)) {
+                this.actions.encounter.set(proxy.slug, proxy);
+            }
```

Comment:
I think it might be better to not do the split into action categories in the actor data, but do it in the sheet data instead.